calour/io.py

```diff
@@ -243,7 +243,11 @@ def _read_metadata(ids, f, kwargs):
         # check if we have duplicate index ids (which will raise)
         dup = metadata.index.duplicated(keep=False)
         if dup.sum() > 0:
-            errmsg = '%d duplicate id values encountered in column %s of mapping file %r:' % (dup.sum(), metadata.index.name, f)
+            column_str = '%d' % index_col
+            if metadata.index.name is not None:
+                if len(metadata.index.name) > 0:
+                    column_str = metadata.index.name
+            errmsg = '%d duplicate id values encountered in index column %s of mapping file %r:' % (dup.sum(), column_str, f)
```

Comment:
why making it so complicated?  only need to say the index column has duplicate IDs? you don't really need the name of index column, right?

Reply:
i think users will find it easier if they get the column name (if it exists) rather than needing to count to column number XXX.
the complication is in the code (and it's not THAT complicated...)
but if you feel strongly to just give the column number instead, can also reduce to that (but the users will be sad....)